moped-toolbox/geometry_prototype/create_prototype.sql

```diff
@@ -0,0 +1,141 @@
+    layer_id_override integer references layers(id) default null, -- think of this layer, which is optional, as an override of the default source, and most likly will always be a pointer to 'drawn_points' or 'drawn_lines'
+    name character varying
+    );
+
+-- feature type generic fields related to the geometry could be put here, plus i really like the global ID space it enforces over feature geometry subtables
+create table features (
+    id serial primary key, 
+    name character varying
+    );
+
+-- should we prefix these tables with something to logically group them? schema them?
+create table signals (
+    signal_id integer default null,
+    geography geography('MULTIPOINT') default null
+    ) inherits (features);
+
+create table sidewalks (
+    sidewalk_id integer default null,
+    sidewalk_name character varying default null,
+    geography geography('MULTILINESTRING') default null
+    ) inherits (features);
+
+create table drawn_points (
+    geography geography('MULTIPOINT') default null
+    ) inherits (features);
+
+create table drawn_lines (
+    geography geography('MULTILINESTRING') default null
+    ) inherits (features);
+
+create table component_feature_map (
+    id serial primary key,
+    component_id integer references components(id),
+    feature_id integer default null
+    );
+
+-- this could be (should be?) a CTE to the project_geography view, but this seems easier to read in a prototype
+create view uniform_features as (
+        select id, 'signals' as table, name, json_build_object('signal_id', signal_id) as attributes, geography
+        FROM signals
+    union all
+        select id, 'sidewalks' as table, name, json_build_object('sidewalk_id', sidewalk_id, 'sidewalk_name', sidewalk_name) as attributes, geography
+        FROM sidewalks
+    union all
+        select id, 'drawn_points' as table, name, null as attributes, geography from drawn_points
+    union all
+        select id, 'drawn_lines' as table, name, null as attributes, geography from drawn_lines
+);
+
+create view project_geography as (
+    select 
+      projects.id as project_id,
+      uniform_features.id as feature_id,
+      components.id as component_id,
+      projects.name as project_name,
+      layers.internal_table as table,
+      layers.reference_layer_primary_key_column as original_fk,
+      components.name as component_name,
+      uniform_features.name as feature_name,
+      uniform_features.attributes as attributes,
+      uniform_features.geography as geography
+    from projects
+    join components on (components.project_id = projects.id)
+    join component_types on (components.component_type_id = component_types.id)
+    join layers on (coalesce(components.layer_id_override, component_types.layer_id) = layers.id)
+    join component_feature_map on (components.id = component_feature_map.component_id)
+    join uniform_features on (component_feature_map.feature_id = uniform_features.id)
+);
```

Comment:
this looks really close. i think the ideal output for a UI client would be an array of "project components", each with an array of geojson (capital F) Features:

```javascript
[
    // this is one project component
    {   
        project_id: 1,
        component_id: 1,
        component_type: {
            component_type: "signal",
            component_subtype: "Traffic"
        },
        subcomponents: [
            "Audible push button"
        ],
        features: [
            { 
                type: "Feature",
                geometry: {
                    type: "MultiLineString"
                    coordinates: [...]
                },
                properties: {
                    id: 1,
                    layer: "segments",
                    segment_id: 12345,
                    street_name: "ALAMO ST",
                }
            },
            { 
                type: "Feature",
                geometry: {
                    type: "MultiLineString"
                    coordinates: [...]
                },
                properties: {
                    id: 2,
                    layer: "drawn_lines",
                }
            }
        ]
    }
]
```

And I want to emphasize **ideal**, because if it makes sense to do some data munging in the editor that is totally fine. we're talking about very straightforward operations on what will always be a fairly small amount of data (at least at the project-level mapping)